File: Some Leetcode and GFG Solutions/242-valid-anagram/242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        return Counter(s)==Counter(t)
#         d={}
#         for i in s:
#             if i not in d:
#                 d[i]=1
#             else:
#                 d[i]+=1
#         for i in t:
#             if i in d:
#                 d[i]-=1
#             else:
#                 d[i]=1
#             if d[i]==0:
#                 d.pop(i)
#         return len(d)==0
            
        

File: CPP/Leetcode/242-valid-anagram/242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        unordered_map<char,int> m1;
        unordered_map<char,int> m2;
        for(int i=0;i<s.length();i++)
        {
           m1[s[i]]++;
        }
        for(int i=0;i<t.length();i++)
        {
           m2[t[i]]++;
        }
        for(int i=0;i<s.length();i++)
        {
           if(m1[s[i]]!=m2[s[i]]) return false;
        }
        for(int i=0;i<t.length();i++)
        {
           if(m1[t[i]]!=m2[t[i]]) return false;
        }
        return true;
    }
};